src/sentry/utils/event_frames.py

```diff
@@ -42,8 +42,12 @@ def java_frame_munger(frame: EventFrame) -> str | None:
     if not frame.filename or not frame.module:
         return None
 
-    if "$$" in frame.module:
-        return frame.module.split("$$")[0].replace(".", "/") + ".java"
+    if "$" in frame.module:
+        path = frame.module.split("$")[0].replace(".", "/")
+        if frame.abs_path and frame.abs_path.count(".") == 1:
+            # Append extension
+            path = path + "." + frame.abs_path.split(".")[-1]
+        return path
```

Comment:
@adinauer, regarding taking advantage of the code in #86280, it may be possible, but it would take some good investigation. I will write it as one of my tasks.